### DISCUSSION TITLE
Sign in github copilot is impossible

### DISCUSSION THREAD

**Discussion by LaurentSauzetOff** [community/community#15026] [Copilot Conversations]
Hello ! 

I have a question. 

I was on the waiting list and recently received the invitation to test Github Copilot. I followed the installation steps on Visual STudio Code but the extension does not ask me to login to Github and the Copilot login icon is disabled. How do I connect the extension to my Github account and thus be able to test Github Copilot ?

**Comment by davecheney** [comment]
Are you using VSCodium?

**Reply by VPaulV** [reply]
Hi! Same problem here VSCodium 1.68.1, Linux system, Firefox 101.0.1. When I press sign in nothing happens. Is there any way to fix this? Reinstalling doesn't help

**Reply by davecheney** [reply]
Sorry, VSCodium is not currently supported. You should use VSCode.

**Reply by VPaulV** [reply]
Is it because of the license issue?

**Reply by davecheney** [reply]
I’m afraid I don’t know the details. All I know is VScodium is currently not compatible with the copilot extension. I imagine this will be addressed at some point, but I don’t have a specific ETA when this might happen.

**Comment by VPaulV** [comment]
Alright guys here is how to make it work in VSCodium, the info is just for educational purpose, just to show that it is theoretically possible and it works great in VSCodium. Please don't break the license agreement.
I use Gentoo, but to adopt the instruction to your system just find the right paths.

1. Install VSCode and VSCodium
```
emerge -av1 vscode
emerge -av vscodium
```
2. Install github copilot in both VSCode and VSCodium
3. Now we need to change the code of copilot a bit. In Gentoo VSCode extensions are in ~/.vscode/extensions/
```
vim ~/.vscode/extensions/github.copilot-1.30.6165/dist/extension.js
```
Find the string with 
```
fetch("https://api.github.com/copilot_internal/v2/token",{headers:{Authorization:
```
And put console.log(t) in front of it. 

So for me was like this:
```
"auth.new_login");const r=await e.get(i.Fetcher).fetch("https://api.github.com/copilot_internal/v2/token",{headers:{Authorization:`token ${t}`}});
```

And I change to:
```
"auth.new_login");console.log(t);const r=await e.get(i.Fetcher).fetch("https://api.github.com/copilot_internal/v2/token",{headers:{Authorization:`token ${t}`}});
```
4. Now open a terminal and start VSCode with
```
vscode --verbose --log trace | grep gho_
```
5. Proceed with sign in to the github using copilot proposed method. Once authorization is finished you will see in the console a string with your token that starts with gho_xxxxx...
6. Copy the token, close VSCode
7. Start VSCodium use sign in method for copilot. It will ask for the token, add the token that you just have copied
8. Now in order to get rid of the wrong vscode version error modify product.json of the VSCodium. On gentoo - /opt/vscodium/resources/app/product.json
```
vim /opt/vscodium/resources/app/product.json
```
Find "GitHub.copilot" and replace two strings with
```
    "GitHub.copilot": ["inlineCompletions", "inlineCompletionsNew", "inlineCompletionsAdditions", "textDocumentNotebook"],
    "GitHub.copilot-nightly": ["inlineCompletions", "inlineCompletionsNew", "inlineCompletionsAdditions", "textDocumentNotebook"],
```
9. Remove VSCode 
```
emerge -avc
```
10. Now it should work

**Reply by Ranguna** [reply]
Unfortunately this is not working for me, I was able to get past the login issues but now I'm getting an error about vscode version "GitHub Copilot requires VS Code 1.57+. Please update your VS Code", with this on the dev tools console:

```
CRITI Extension 'github.copilot' appears in product.json but enables LESS API proposals than the extension wants.
package.json (LOSES): inlineCompletionsNew, inlineCompletionsAdditions
product.json (WINS): 

log.ts:313   ERR [GitHub.copilot]: editor/inlineCompletions/actions is a proposed menu identifier. It requires 'package.json#enabledApiProposals: ["inlineCompletionsAdditions"]' and is only available when running out of dev or with the following command line switch: --enable-proposed-api GitHub.copilot
```

I've added copilot runtime arg and updated product.json with the required proposals, nothing worked. Thanks a lot microsoft.

**Reply by VPaulV** [reply]
Did you try running vscodium with --enable-proposed-api GitHub.copilot as suggested in your error message?

**Reply by Ranguna** [reply]
@VPaulV yep, same errors pop up. Is this still working for you ?

Which version of vscodium and copilot do you have ?

**Reply by VPaulV** [reply]
@Ranguna, yes using it on a daily basis, everything is good. Maybe also try to add package.json#enabledApiProposals: ["inlineCompletionsAdditions"] ? 
```
vscodium -v
1.68.1
30d9c6cd9483b2cc586687151bcbcd635f373630
x64
```

**Reply by Ranguna** [reply]
@VPaulV  I have the same commit. Copilot v1.31.6194 ?

`package.json#enabledApiProposals: ["inlineCompletionsAdditions"]` in the product.json ?

Will give it a try.

**Reply by Ranguna** [reply]
Hey it worked!

Looks like my product.json was being overwritten and didn't have the correct features enabled for copilot. Copied your configs again and it all worked out!

Thanks for the help :+1:

**Reply by VPaulV** [reply]
Alright! Great to know, thank you for testing, I will update the instruction now

**Reply by Ranguna** [reply]
Your instructions are good. There's no need to add the `package.json#enabledApiProposals: ["inlineCompletionsAdditions"]` line, the ones you have are enough :+1:

**Reply by VPaulV** [reply]
I would not be so angry on Microsoft tho. Copilot is great, vscode as well. It is their product and it is understandable that they want to focus on their editor not some forks. I think once copilot will get out of beta we will find out some better ways to do it

**Reply by Ranguna** [reply]
True. Their products are pretty good, my biggest bief is with their communication and lack of transparency. Either way, as long as the community is there to save the day, it's all good.
Once again, thanks for the help!
